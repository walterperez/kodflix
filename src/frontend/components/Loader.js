import React from "react";

export default function Loader () {
    return (
        <div className="flex-container-loader">
            <div className="loader"></div>
        </div>
    )
};