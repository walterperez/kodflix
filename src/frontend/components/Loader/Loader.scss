/* Loader Component */
.loader {
  width: 40px;
  height: 40px;
  border: 15px solid grey;
  border-top: 15px solid rgba(209, 209, 5, 1);
  border-radius: 50%;
  animation: spin 2s linear infinite;
}
@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.flex-container-loader {
  width: 100%;
  min-height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}
