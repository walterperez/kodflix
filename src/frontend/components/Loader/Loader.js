import React from 'react';
import './Loader.scss';

export default function Loader() {
  return (
    <div className="flex-container-loader" data-test="Loader">
      <div className="loader" />
    </div>
  );
}
