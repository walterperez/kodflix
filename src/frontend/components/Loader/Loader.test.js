import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Loader from './Loader';

configure({ adapter: new Adapter() });

describe('Loader', () => {
  it('Render Snapshot', () => {
    const output = renderer.create(<Loader />);
    expect(output).toMatchSnapshot();
  });

  it('Render', () => {
    const wrapper = shallow(<Loader />);
    const loader = wrapper.find(`[data-test='Loader']`);
    expect(loader.length).toBe(1);
  });
});
