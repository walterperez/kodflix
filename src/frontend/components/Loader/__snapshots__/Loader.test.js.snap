// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Render Snapshot 1`] = `
<div
  className="flex-container-loader"
  data-test="Loader"
>
  <div
    className="loader"
  />
</div>
`;
